scipy/stats/_stats_py.py

```diff
@@ -10822,7 +10810,10 @@ def _xp_mean(x, /, *, axis=None, weights=None, keepdims=False, nan_policy='propa
 
     # Perform the mean calculation itself
     if weights is None:
-        return xp.mean(x, axis=axis, keepdims=keepdims)
+        if is_array_api_strict(xp) and xp.isdtype(x.dtype, 'complex floating'):
```

Comment:
Should we just skip strict rather than special case since it looks like this is just waiting for a new release? https://github.com/data-apis/array-api-strict/pull/110

Reply:
Sure, I can make the changes (or you can commit https://github.com/scipy/scipy/pull/22405#pullrequestreview-2574411591) if the rest looks ready.